lib/modules/manager/pip-compile/common.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { allowedPipOptions, extractHeaderCommand } from './common';
+
+function getCommandInHeader(command: string) {
+  return `#
+# This file is autogenerated by pip-compile with Python 3.11
+# by the following command:
+#
+#    ${command}
+#
+`;
+}
+
+describe('modules/manager/pip-compile/common', () => {
+  describe('extractHeaderCommand()', () => {
+    it.each([
+      '-v',
+      '--generate-hashes',
+      '--resolver=backtracking',
+      '--resolver=legacy',
+      '--output-file=reqs.txt',
+    ])('returns object on correct options', (argument: string) => {
+      expect(
+        extractHeaderCommand(
+          getCommandInHeader(`pip-compile ${argument} reqs.in`),
+          'reqs.txt',
+        ),
+      ).toBeObject();
+    });
+
+    it.each(['--resolver', '--output-file reqs.txt', '--extra = jupyter'])(
+      'errors on malformed options with argument',
+      (argument: string) => {
+        expect(() =>
+          extractHeaderCommand(
+            getCommandInHeader(`pip-compile ${argument} reqs.in`),
+            'reqs.txt',
+          ),
+        ).toThrow(/equal sign/);
+      },
+    );
+
+    it.each(['--foo', '-x', '--$(curl this)', '--bar=sus'])(
+      'errors on unknown options',
+      (argument: string) => {
+        expect(() =>
+          extractHeaderCommand(
+            getCommandInHeader(`pip-compile ${argument} reqs.in`),
+            'reqs.txt',
+          ),
+        ).toThrow(/not supported/);
+      },
+    );
+
+    it.each(['--no-header'])(
+      'always errors on not allowed options',
+      (argument: string) => {
+        expect(() =>
+          extractHeaderCommand(
+            getCommandInHeader(`pip-compile ${argument} reqs.in`),
+            'reqs.txt',
+          ),
+        ).toThrow(/not allowed/);
+      },
+    );
+
+    test('error when no source files passed as arguments', () => {
```

Comment:
```suggestion
    it('error when no source files passed as arguments', () => {
```
please be consistent